raspberry_pi_revision.c

```diff
@@ -455,13 +470,59 @@ getRaspberryPiRevision()
 
 //-------------------------------------------------------------------------
 
+int
+getRaspberryPiExtendedRevision()
+{
+    int raspberryPiExtendedRevision = 0;
+
+    FILE *fp = fopen("/proc/device-tree/chosen/rpi-boardrev-ext", "r");
+
+    if (fp == NULL)
+    {
+        perror("/proc/device-tree/chosen/rpi-boardrev-ext");
+        return raspberryPiExtendedRevision;
+    }
+
+    uint8_t data[4];
+    uint32_t value = 0;
+
+    size_t numRead = fread(data, 1, 4, fp);
+    if (numRead != 4)
+    {
+        perror("/proc/device-tree/chosen/rpi-boardrev-ext");
```

Comment:
This error could be more expressive. The file has been opened successfully, but we were unable to read the required number of bytes.